File: src/main/java/com/example/NeptuneApp.java
package com.example;

import io.github.plume.oss.Extractor;
import io.github.plume.oss.drivers.DriverFactory;
import io.github.plume.oss.drivers.GraphDatabase;
import io.github.plume.oss.drivers.NeptuneDriver;

import java.io.File;
import java.io.IOException;

public class NeptuneApp {

    public static void main(String[] args) {
        System.out.println("Creating driver");
        try (NeptuneDriver driver = (NeptuneDriver) DriverFactory.invoke(GraphDatabase.NEPTUNE)) {
            driver.addHostnames("<neptune-cluster-address-without-https>")
                    .keyCertChainFile("<pem-certificate-path-here>")
                    .port(8182)
                    .connect();
            // Create the extractor with the driver and class root directory
            Extractor extractor = new Extractor(driver, new File("./src/main/resources/examples"));
            File f = new File("./src/main/resources/examples/intraprocedural/basic/Basic1.java");
            // Load the extractor with the directory of all the tests
            extractor.load(f);
            // Project the loaded classes to the graph database
            System.out.println("Projecting graph");
            extractor.project();
            driver.close();
            System.out.println("Done!");
            System.exit(0);
        } catch (IOException e) {
            System.out.println("Something went wrong! Details " + e.getMessage());
        }
    }

}

File: src/main/scala/com/example/NeptuneApp.scala
package com.example

import com.github.plume.oss.Jimple2Cpg
import com.github.plume.oss.drivers.NeptuneDriver

import scala.util.Using

object NeptuneApp {

  def main(args: Array[String]): Unit = {
    val targetDir = "./example"

    println("Creating driver")
    Using.resource(new NeptuneDriver(
      hostname = "<neptune-cluster-address-without-https>",
      port = 8182,
      keyCertChainFile = "<pem-certificate-path-here>"
    )) { d =>
      println(s"Creating CPG from .class files found under $targetDir")
      new Jimple2Cpg().createCpg(rawSourceCodePath = targetDir, driver = d)
    }
    println(s"Done! CPG persisted on the database")
  }

}
